packages/platform/src/internal/fileSystem.ts

```diff
@@ -124,3 +127,32 @@ const stream = (file: File, {
     { capacity: bufferSize }
   )
 }
+
+/** @internal */
+export const logger = (
+  path: string,
+  options?: OpenFileOptions
+) =>
+  Effect.gen(function*(_) {
+    const fs = yield* _(tag)
+    const logFile = yield* _(fs.open(path, options))
+    const scope = yield* _(Scope.Scope)
+    const queue = yield* _(Queue.unbounded<string>())
+
+    yield* _(
+      Effect.forkIn(
+        Stream.runForEachChunk(Stream.fromQueue(queue), (chunk) =>
+          logFile.write(
+            new TextEncoder().encode(Chunk.join(chunk, "\n") + "\n")
+          )),
+        scope
+      )
+    )
+
+    yield* _(Effect.yieldNow())
+
+    return Logger.make((options) => {
+      const formatted = Logger.stringLogger.log(options)
```

Comment:
I think the logger should be passed in rather than assume the format.

And maybe move to a PlatformLogger module.

Reply:
Updated. I kind of like having a `format?: (_: Logger.Logger.Options<unknown>) => string` more than passing in a Logger though with the option to pass in for example `{ format: Logger.stringLogger.log }`. Its more flexible.

Reply:
I generally prefer composition over options if possible. I had a play around and I think something like this is nice: https://github.com/Effect-TS/effect/commit/ed0d513fb5c8f14d588553cbe8197e684a69e60b#diff-8f9519c9651eac0a7c450530c155b62fe4f248e54db12f7ac0c62fb90b1be189R5

Reply:
Also found a bug that needs addressing - if the program shuts down logs can be dropped. You can see the test case in the branch above.

Reply:
Ah yeah, that's definitely better. Composition is a lot nicer in this case.

Reply:
> Also found a bug that needs addressing - if the program shuts down logs can be dropped. You can see the test case in the branch above.

Switching to a runtime based approach fixes the issue: https://github.com/Effect-TS/effect/compare/file-logger-composition#diff-a5d9b91049b63b5c55abf4414eea8eacd153c692a23a3e03e9bfb11b4d7ef686R36

Reply:
Oh nice. It seems to work well. It's even simpler w/o the queue. I wonder what would happen if logs came in faster than they could be written though.